docs/explanations/performance.md

```diff
@@ -0,0 +1,124 @@
+# Napari Performance
+
+Performance is a core feature of napari. Performance is a concern for every
+feature that napari implements. Without adequate performance users will migrate
+to other visualization tools, even if napari has impressive functionality.
```

Comment:
I would frame this a bit differently. napari has been successful so far because it solves use cases that nobody had adequately solved in Python, even if some of its design is failing to scale so far. So here I would mention more intrinsic characteristics of performance as it relates to visualisation:

- Fundamental use cases such as data exploration depend on performance, because one needs to be able to swiftly find interesting parts of the data. If things are slow, one drowns in noise.
- With visualisation and UI in particular, and in contrast to analysis code, fast, responsive software is joyful to use, while slowdowns are frustrating and makes one want to seek alternatives or simply stop using visualisation altogether, opting instead for summary statistics. This results in loss of information and missed opportunities.

Reply:
That intro wasn't in my original doc. I just tacked it on. Your suggests were good I revised it to try and capture the idea that performance for "interactive" tools even super criticial.